### DISCUSSION TITLE
How to globally catch unhandled exceptions on different platforms

### DISCUSSION THREAD

**Discussion by Daoting** [unoplatform/uno#9390] [Q&A] [answered]
I used the following method to catch unhandled exceptions for Xamarin and UWP and it works fine:
```
#if WIN
            Application.Current.UnhandledException += OnUwpUnhandledException;
#elif ANDROID
            Android.Runtime.AndroidEnvironment.UnhandledExceptionRaiser += OnAndroidUnhandledException;
#elif IOS
            ObjCRuntime.Runtime.MarshalManagedException += (_, args) =>args.ExceptionMode = ObjCRuntime.MarshalManagedExceptionMode.UnwindNativeCode;;
            AppDomain.CurrentDomain.UnhandledException += (s, e) => OnUnhandledException(e.ExceptionObject as Exception);
#elif WASM
            //TaskScheduler.UnobservedTaskException += (s, e) => OnUnhandledException(e.Exception);
            AppDomain.CurrentDomain.UnhandledException += (s, e) => OnUnhandledException(e.ExceptionObject as Exception);
#endif
```

After upgrading to .net6, it was never captured, although many methods have been tried. However, there is an exception log in the Output window. Is it handled by uno? The app no longer has any unhandled exceptions, but I need it.
```
[DOTNET] [41m[30mfail[39m[22m[49m: Microsoft.UI.Xaml.UIElement[0]
[DOTNET]       Failed to raise 'PointerReleasedEvent': Dt.Core.KnownException: business warning
[DOTNET]          at Dt.Core.Throw.ThrowMsg(String p_msg) in D:\Dt\Service\src\Dt.Core\Internal\Misc\Throw.cs:line 72
[DOTNET]          at Dt.Core.Throw.Msg(String p_msg) in D:\Dt\Service\src\Dt.Core\Internal\Misc\Throw.cs:line 62
[DOTNET]          at Dt.Sample.NotifyDemo.OnKnownException(Object sender, RoutedEventArgs e) in D:\Dt\Client\Dt.Sample\Structure\NotifyDemo.xaml.cs:line 102
```
Similar to this https://github.com/unoplatform/uno/discussions/9202

Thanks.

**Comment by Daoting** [comment]
@jeromelaban

Thank you for answer this question in your busy schedule!
It was also not found in the documentation.

**Comment by Daoting** [comment, marked as answer]
@jeromelaban

How to catch unhandled exceptions globally on different platforms is an important question for us.
Thanks!

**Reply by jeromelaban** [reply]
Please do not repeat questions. .NET 6 is something to ask the .NET team, as Uno does not interfere with the way the runtime works.

**Reply by Daoting** [reply]
Sorry, I thought you didn't see it.

**Comment by SongOfYouth** [comment]
Any unhandled exception will crash the app now on wasm and android, but it works on winui and desktop! I want to handle the exception globaly on my app, such as show it via unified message box. @jeromelaban
